main/inc/lib/sessionmanager.lib.php

```diff
@@ -5429,4 +5428,266 @@ public static function getCoursesListByCourseCoach($coachId)
+                        // Create an array to index ids to session_id
+                        $sessionCourseListBySession[$sessionCourse['session_id']][] = $key;
+                    }
+                }
+                // Join lists
+                if (is_array($sessionList)) {
+                    foreach ($sessionList as $id => &$row) {
+                        if (
+                            !empty($sessionFieldValueListBySession) &&
+                            is_array($sessionFieldValueListBySession[$id])
+                        ) {
+                            // If have an index array for session extra fields, use it to join arrays
+                            foreach ($sessionFieldValueListBySession[$id] as $key) {
+                                $row['extra'][$key] = array(
+                                    'field_name' => $sessionFieldList[$sessionFieldValueList[$key]['field_id']]['field_variable'],
+                                    'field_value' => $sessionFieldValueList[$key]['field_value'],
+                                );
+                            }
+                        }
+                        if (
+                            !empty($sessionCourseListBySession) &&
+                            is_array($sessionCourseListBySession[$id])
+                        ) {
+                            // If have an index array for session course coach, use it to join arrays
+                            foreach ($sessionCourseListBySession[$id] as $key) {
+                                $row['course'][$key] = array(
+                                    'course_id' => $sessionCourseList[$key]['course_id'],
+                                    'course_code' => $sessionCourseList[$key]['course_code'],
+                                    'course_title' => $sessionCourseList[$key]['course_title'],
+                                    'coach_username' => $sessionCourseList[$key]['coach_username'],
+                                    'coach_firstname' => $sessionCourseList[$key]['coach_firstname'],
+                                    'coach_lastname' => $sessionCourseList[$key]['coach_lastname'],
+                                );
+                            }
+                        }
+                    }
+                }
+
+                return $sessionList;
+            } else {
+                // Not found result, update error message
+                $errorResult['errorMessage'] = 'Not found any session for session category id ' . $sessionCategoryId;
+            }
+
+            return $errorResult;
+        }
+    }
+
+    /**
+     * Return session description from
+     * @param int $sessionId
+     * @return string
+     */
+    public static function getDescriptionFromSessionId($sessionId)
+    {
+        // Init variables
+        $sessionId = intval($sessionId);
+        $description = '';
+        // Check if session id is valid
+        if ($sessionId !== 0) {
```

Comment:
should be > 0 (given session ID is a int unsigned, it doesn't hold negative values)